syntax = "proto3";

package boltz;

// Common message types for all services

// Request for job status
message JobStatusRequest {
  string job_id = 1;
}

// Response for job status
message JobStatusResponse {
  string job_id = 1;
  string status = 2;
  float progress = 3;
  string result_path = 4;
  string error_message = 5;
}

// Request to cancel job
message CancelJobRequest {
  string job_id = 1;
}

// Response for job cancellation
message CancelJobResponse {
  string job_id = 1;
  string status = 2;
  string error_message = 3;
}
